src/saml2/client_base.py

```diff
@@ -181,6 +181,9 @@ def __init__(self, config=None, identity_cache=None, state_cache=None,
 
             setattr(self, attr, val)
 
+        for algorithm in ('signing_algorithm', 'digest_algorithm'):
+            setattr(self, algorithm, self.config.getattr(algorithm, "sp"))
```

Comment:
I would prefer the more explicit
```python
self.signing_algorithm = self.config.getattr(signing_algorithm, "sp")
self.digest_algorithm = self.config.getattr(digest_algorithm, "sp")
```
but I see that some lines above is the `setattr(self, attr, val)` so maybe it's just me :man_shrugging: